Xamarin.Forms.Core.UnitTests/ShellTests.cs

```diff
@@ -800,5 +800,25 @@ public async Task NavigatedFiresAfterContentIsCreatedWhenUsingTemplate()
 			Assert.AreEqual("//rootlevelcontent1/details", shell.CurrentState.Location.ToString());
 		}
 
+		[Test]
+		public async Task FailWhenNoShellContentProvided()
+		{
+			var shell = new Shell();
+			var tabBar = new TabBar();
+			var tab = new Tab();
+			tabBar.Items.Add(tab);
+			shell.Items.Add(tabBar);
+			Assert.Catch(typeof(InvalidOperationException), () => Routing.RegisterRoute(string.Empty, typeof(ContentPage)));
```

Comment:
@techduggu these are failing. I'm not following how these are supposed to trigger an InvalidOperationException?

Since the InvalidOperationException is only being triggered by platform code it's not going to trigger these exceptions from the unit tests.

Reply:
@PureWeen I checked these and it is strange that it's now failing for `ArgumentException` too (the previous state). I'm sure the Xamarin build was working fine earlier (at least with ArgumentException) but not anymore. The exception we get while running the test is that - _it expects an instance of <System.ArgumentException> (or InvalidOperationException) But was: null._

It is clear that exceptions being triggered by platform code, so Should we just remove these Unit test cases?